tests/mir-opt/inline/caller_with_trivial_bound.rs

```diff
@@ -18,6 +17,8 @@ pub fn foo<T>()
 where
     IntFactory: Factory<T>,
```

Comment:
```suggestion
    // Because of this trivial bound, the inliner fails to normalize `<IntFactory as Factory<T>>::Item`.
    // Verify that we do not inline anything, which would cause validation ICEs.
    IntFactory: Factory<T>,
```